Python/symtable.c

```diff
@@ -558,6 +559,65 @@ analyze_name(PySTEntryObject *ste, PyObject *scopes, PyObject *name, long flags,
     return 1;
 }
 
+static int
+is_free_in_children(PySTEntryObject *entry, PyObject *key) {
+    for (Py_ssize_t i = 0; i < PyList_GET_SIZE(entry->ste_children); i++) {
+        PySTEntryObject *child_ste = (PySTEntryObject *)PyList_GET_ITEM(
+            entry->ste_children, i);
+        long scope = _PyST_GetScope(child_ste, key);
+        if (scope == FREE) {
+            return 1;
+        }
+    }
+    return 0;
+}
+
+static int
+inline_comprehension(PySTEntryObject *ste, PySTEntryObject *comp,
+                     PyObject *scopes, PyObject *comp_free)
+{
+    PyObject *k, *v;
+    Py_ssize_t pos = 0;
+    while (PyDict_Next(comp->ste_symbols, &pos, &k, &v)) {
+        // skip comprehension parameter
+        long comp_flags = PyLong_AS_LONG(v);
```

Comment:
we don't need to check for error here?

Reply:
No, I don't think we need to. The only error case for `PyLong_AS_LONG` is "doesn't fit in a long" and it's not possible for a variable's scope flags to have such a large value. There are lots of already-existing similar uses of `PyLong_AS_LONG` in `symtable.c`, and none of them check for error.